services/site/src/app/challenge/[challengeSlug]/page.tsx

```diff
@@ -0,0 +1,85 @@
+import Footer from "app/components/Footer";
+import Check from "app/components/icons/Check";
+import Navigation from "app/components/Navigation";
+import { ChallengeDetailsBySlugDocument, ChallengeDetailsBySlugQuery, ChallengeDetailsBySlugQueryResult, LevelStatus } from "app/generated/graphql";
+import { getApolloClient } from "app/lib/apollo-client/rsc";
+import clsx from "clsx";
+import { Metadata } from "next";
+import Link from "next/link";
+import { notFound } from "next/navigation";
+import sanitizeHtml from "sanitize-html";
+
+type PageProps = {
+  params: {
+    challengeSlug: string;
+  };
+};
+
+async function getChallenge(slug: string): Promise<ChallengeDetailsBySlugQueryResult["data"]["challenge"]> {
+  const client = getApolloClient();
+
+  const response = await client.query<ChallengeDetailsBySlugQuery>({
+    query: ChallengeDetailsBySlugDocument,
+    variables: { slug },
+  });
+
+  return response.data.challenge;
+}
+
+const Challenge = async ({ params: { challengeSlug } }: PageProps): Promise<React.ReactElement> => {
+  const challenge = await getChallenge(challengeSlug);
+
+  if (!challenge) {
+    notFound();
+  }
+
+  const firstUnfinishedLevel =
+    challenge === undefined
+      ? undefined
+      : challenge.levels.find((level) => level.status === LevelStatus.Open || level.status === LevelStatus.InProgress);
+
+  return (
+    <>
+      <Navigation displayBreadcrumbs />
+      <main className="h-full box-border max-w-screen-3xl mx-auto">
+        <div className="mx-8 py-8 h-main max-w-screen-3xl sm:mx-12 lg:mt-12 lg:mx-24">
+          <h1 className="mb-8 h2 md:h1">{challenge?.name}</h1>
```

Comment:
I would add a SR text like "Challenge:" to indicate would are on the challenge detail view. otherwise the heading just says "heading" for the headings challenge.